src/Agent.Listener/SelfUpdater.cs

```diff
@@ -178,20 +178,14 @@ private async Task<bool> HashValidation(string archiveFile)
                 return true;
             }
 
-            bool isHostedServer;
             await _serverUtil.DetermineDeploymentType(_serverUrl, _creds, _locationServer);
```

Comment:
Can we get rid of serverUtil usage here then?

Reply:
Looks like yes, serverUtil can be removed from here. I have removed this from SelfUpdater by new commit. And no one issue was detected during testing.
But I'm going to add Denis, Alexander and Daniil to the PR as reviews. They are author and reviewers in PR with adding of serverUtil to SelfUpdater (https://github.com/microsoft/azure-pipelines-agent/pull/3524).